### ISSUE TITLE
Lab 03 - Issue with X and Y axis cutting off text

### ISSUE THREAD

**Issue by pharri14** [DS4PS/cpp-526-sum-2020#14]
**I am having issues with not having the X and Y axis cut off the text/data.**

This is what the code looks like right now

```{r}
plot.new()
plot.window( xlim = c(1900, 2012),
             ylim = c(0, 10) )

# X-axis
axis( side = 1,
      at = seq( from = 1900,
                to = 2012,
                by = 10),
      col = "black",
      cex.axis = 0.7)

# Y-axis
axis( side = 4,
      las = 1, 
      at = seq( from = 0,
                to = 9,
                by = 1),
      cex.axis = 0.7,
      col = NA, 
      col.axis = "gray50",
      font = 2)
```
Here is a small snippet of the graph. The code is producing a graph that cuts off the text (at the top of the screenshot) and is also not aligning the X-axis with zero (at the bottom of the screenshot). **Does anyone know what lines of code I am missing that could fix these issues?**

![Screen Shot 2020-06-04 at 1 25 18 PM](https://user-images.githubusercontent.com/65568459/83806737-d7770d00-a666-11ea-8cc7-016fd0034b13.png)

**Comment by jamisoncrawford** [comment]
Hi @pharri14, sorry for the delay on this - I'll be back to checking my emails daily after this past week.

Rather than modifying the y-axis in `axis()`, modify the arguments in your call to `text()` to include `xpd = NA`.  This allows the text to overflow rather than get cut off.

Let me know if this works for you!

**Comment by malmufre** [comment]
Hi @jamisoncrawford and @pharri14  ,
I had the same issue and I tried fixing it by using ```text()``` and including``` xpd=Na``` however It only allowed  the text "League average" to overflow. The X-axis is still far away from the Y-axis unlike the graph.
Any Idea how to bring the x-axis closer to the Y-axis?
Thanks!

**Comment by gzbib** [comment]
Hello Sir,

I am having the same issue and I tried searching for a function to extend or stretch x-axis but I didn't find any.

Thanks in advance.

**Comment by jamisoncrawford** [comment]
Hi ya'll - sorry for the delay.  There are a few different ways to do this, depending on the result you want.  The key terms, if you'd like to find more about this on sites like *Stack Overflow*, include "margin", "plot", "axis", and "adjust" - just in case the following isn't helpful.

### Adjusting Margins

Before calling `plot()` or related functions, set your graphical parameters with function `par()`.  Inside `par()`, you'll be most interested in these arguments:

* `mar =`: Accepts four numeric values contained within `c()`, e.g. `c(2, 4, 2, 4)`, corresponding to the margin sizes for bottom, left, top, and right margins, respectively
* `mgp =`: Accepts three numeric values in `c()` for the location of the axes labels, tick labels, and ticks, respectively

Remember to experiment with negative numeric values to adjust positions, as well (**[Source](http://rfunction.com/archives/1302)**).

### Making "Y = 0" Meet X-Axis

I was able to find a helpful answer on *Stack Overflow* **[here](https://stackoverflow.com/questions/39042661/how-to-align-the-x-axis-on-y-0-in-r)**.

In short, in your call to `plot()`, use argument and parameter `yaxs = "i"`.  Does this do the trick?

**Comment by jamisoncrawford** [comment]
@lecy tagging Dr. Lecy since he has much more mastery over core R graphics and I typically use `ggplot2`.  Does this seem like the right way to go to resolve this?

**Comment by lecy** [comment]
To prevent cutting off text you need to create a buffer zone at the end of the graph. If the text is on the graph (on the canvas part) the best thing to do is just extend the x-axis by z units (10 years in this case?).

xlim sets the x limits. min and max:

```
plot( ..., xlim=c( min(x), max(x) + z ) )
```
Otherwise you would have to add text to the margin (region outside the canvas but not the axis) using `mtext()`, but that is more complicated. 

~~~

@malmufre 

I'm not sure what you mean by moving the x-axis closer to the y-axis, but I suspect you are referring to tick marks? 

It's not a trivial matter to automatically generate meaningful tick marks, but R does a pretty good job usually. It will find the range, then try to split it into 4 or 5 reference points that are numbered on the graph. 

You can override this process by setting tick marks manually with the axis function. 

```r
plot( 1:10, 1:10, xaxt="n", xlim=c(0,15), bty="n" )
axis( side=1, at=c(0,2,4,6,8,10) )
text( x=10, y=3, labels="Some extra room for text" )
```

![image](https://user-images.githubusercontent.com/1209099/84020783-3e7f1500-a938-11ea-8308-fff26542e01a.png)
